Runtime/Model/BacktraceUnityMessage.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Globalization;
+using System.Text;
+using UnityEngine;
+
+namespace Backtrace.Unity.Model
+{
+    /// <summary>
+    /// Unity message representation in Backtrace-Unity library
+    /// </summary>
+    internal class BacktraceUnityMessage
+    {
+        private readonly string _formattedMessage;
+        public readonly string Message;
+        public readonly string StackTrace;
+        public readonly LogType Type;
+
+        public BacktraceUnityMessage(BacktraceReport report)
+        {
+            if (report == null)
+            {
+                throw new ArgumentException("report");
```

Comment:
BacktraceUnityMessage constructor can throw but we don't do exception handling when we construct BacktraceUnityMessage throughout the code

Reply:
I think it's expected situation - the reason behind this decision is to let the developer know about mistakes when the developer creates new functionality around BacktraceUnityMessage. Our code will never throw this exception - all reports are defined and there won't be an option in our API that the report object will be undefined. 

This conditions check future improvements to make sure we have valid information required to create a Unity message that we can change into the specific log.

Reply:
Sounds good, thank you!